File: src/paths.R
library(yaml)

# Works regardless of where the script sits in the folder tree
ROOT <- here::here()  # install.packages("here") if needed

PATHS <- yaml::read_yaml(file.path(ROOT, "config/paths.yaml"))

# Helper function — equivalent of the Python get()
get_path <- function(chapter, key) {
  file.path(ROOT, PATHS[[chapter]][[key]])
}

File: src/paths.py
import yaml
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent

with open(ROOT / "config/paths.yaml") as f:
    PATHS = yaml.safe_load(f)


def get(chapter: str, key: str) -> Path:
    """Return an absolute path for a given chapter and key."""
    return ROOT / PATHS[chapter][key]
